app/controllers/video-toolbar.controller.js

```diff
@@ -0,0 +1,58 @@
+(function() {
+  'use strict';
+
+  class VideoToolbarCtrl {
+    constructor(StorageService, VideoService, FilterService) {
+      Object.assign(this, {StorageService, VideoService, FilterService});
+      this.videos = this.VideoService.videos;

```

Comment:
to co mi sie generalnie nie podoba w tym kodzie jest swietnie ujete w tym miejscu. jest to najwazniejsza uwaga dotyczaca calej architektury i podejscia FRP. takze wazne, zebys to zrozumiala dobrze:

przez to, ze tworzysz tu referencje do danych z zewnatrz Twoj element przestaje byc elementem bezstanowym. videotoolbar juz nie jest stateless component - operuje na aktualnym stanie aplikacji (dlatego, ze nie masz pobrania wartosci, pozniej przerabiania ich a pozniej wywolania callbacku z akcja do wykonania). 

w dodatku Twoj komponent przez te zaleznosci jest nieprzenaszalny.

zastanow sie jak powinnas napisac kod, zeby zrobic aplikacja taka jak:
. komponent strony - stanowy
. serwisy - w tej aplikacji zrobilbym je bezstanowe
. kazdy komponent w kodzie html i odpowiadajace mu kontrolery (dzialajace jak dyrektywy) - bezstanowe

w tym wypadku VideoListCtrl (komponent strony) powinien byc stanowy, a np. VideoToolbarCtrl (komponent analogiczny do dyrektywy) - powinien byc bezstanowy.

bedziesz miala do poprawy duzo rzeczy, kilka spraw w tym pomocnych:
. pamietaj, ze: wartosci powinny byc przekazywane z gory do dolu przez bindings (z parametrem '<'), callbacki do komponentow bardziej przez bindings ('&') niz przez dependency injection

. jezeli gdziekolwiek masz:
```javascript
this.someVarFromCtrl = this.SomeService.someObjValueFromService; // nastepuje referencja a nie przypisanie
// swoja droga nazwy serwisow powinny byc zapisywane mala litera w tym miejscu juz, bo juz jest to konkretny obiekt a nie konstruktor
```
to musisz to zrefaktorowac i albo przekazac wartosc przez bindings albo uzyc jakiegos gettera z service'u (np. this.videoService.getAll() - najczesciej przekazujac parametr, np. id do zadan typu this.videoService.get(id))

. metody powinny zawsze byc wywolywane z argumentem (jezeli widzisz w kodzie, ze jest inaczej, to zazwyczaj jest blad)

. jezeli pobierasz jakies dane zazwyczaj powinnas robic to przy uwzglednieniu asynchronicznosci i zastosowac promise